### PR TITLE
Deploy cartridge doc to test server

### LINKED ISSUE #1391: [4pt] Deploy feature branches of Cartridge docs to tarantool-develop

### PR DESCRIPTION
**Pull request by artembo**
Add deploy-branch workflow for reviewing documentation on the test server

Close #1391

### CODE REVIEW

File: .github/workflows/deploy-branch.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name: Deploy-branch
+
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - 'rst/**.rst'
```
Comment (rosik) [inline .github/workflows/deploy-branch.yml:7]: Seems suspicious.

File: .github/workflows/deploy-branch.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+name: Deploy-branch
```
Comment (rosik) [inline .github/workflows/deploy-branch.yml:1]: I don't like the name

File: .github/workflows/deploy-branch.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Set branch name from source branch
+        run: echo "BRANCH_NAME=${GITHUB_HEAD_REF##*/}" >> $GITHUB_ENV
+
+      - name: Start dev server deployment
+        uses: bobheadxi/deployments@v0.5.2
```
Comment (rosik) [inline .github/workflows/deploy-branch.yml]: Why do you use it, and not https://github.com/marketplace/actions/deployment-action?
(Describe it in PR comments, please)

**Comment by artembo** [comment]
Better alternative - https://github.com/tarantool/doc/issues/2155

**Pull request closed without merging**